src/ramanchada2/io/HSDS.py

```diff
@@ -67,18 +67,22 @@ def write_nexus(filename: str,
 class DatasetExistsError(Exception):
     pass
 
+def sanitize_key(key: str) -> str:
+    return ''.join(c if ord(c) < 128 else '_' for c in key)
```

Comment:
Instead of `ord(c) < 128`, maybe use [`c.isascii()`](https://docs.python.org/3/library/stdtypes.html#str.isascii) or even `c.isalnum()` if the key will be only alphanumeric?

Reply:
isascii should be fine. 
there are some .spc with metadata keys with specific unicode that are breaking hdf5 writing

Reply:
we don't need alphanum only, keys can contain brackets, dots, etc

Reply:
If we know what the valid characters are, maybe test for them? Because currently we would also allow some non-printable characters, e.g. ^G, BELL.